File: location-tracker-api/src/main/java/com/gmail/kadoshnikovkirill/locationtracker/repository/cache/LocationCache.java
package com.gmail.kadoshnikovkirill.locationtracker.repository.cache;

import com.gmail.kadoshnikovkirill.locationtracker.dto.LocationDto;
import reactor.core.publisher.Mono;

public interface LocationCache {

    Mono<LocationDto> get(float lat, float lon);

    void put(float lat, float lon, LocationDto dto);
}

File: location-tracker-api/src/main/java/com/gmail/kadoshnikovkirill/locationtracker/repository/cache/LocationCache.kt
package com.gmail.kadoshnikovkirill.locationtracker.repository.cache

import com.gmail.kadoshnikovkirill.locationtracker.dto.LocationDto
import reactor.core.publisher.Mono

interface LocationCache {
    operator fun get(lat: Float, lon: Float): Mono<LocationDto>
    operator fun set(lat: Float, lon: Float, dto: LocationDto)
}